logging/cloudwatch/hook.go

```diff
@@ -173,11 +172,21 @@ func (h *Hook) sendBatch(batch []*cloudwatchlogs.InputLogEvent) {
 		SequenceToken: h.nextSequenceToken,
 	}
 	resp, err := h.svc.PutLogEvents(params)
-	if err != nil {
-		h.err = &err
-	} else {
-		h.nextSequenceToken = resp.NextSequenceToken
+	if err == nil {
+		h.NextSequenceToken = resp.NextSequenceToken
+		h.m.Unlock()
+		return
+	}
+
+	h.err = &err
+	if aerr, ok := err.(*cloudwatchlogs.InvalidSequenceTokenException); ok {
+		h.nextSequenceToken = aerr.ExpectedSequenceToken
+		h.m.Unlock()
+		h.sendBatch()
```

Comment:
I'm not sure where the sendBatch() method that doesn't take any arguments is located.  Are you intending to call sendBatch() recursively with the batch that failed to send?